new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/LoggingExporter.java

```diff
@@ -0,0 +1,474 @@
+import oracle.weblogic.kubernetes.actions.impl.primitive.Kubernetes;
+import oracle.weblogic.kubernetes.assertions.impl.Deployment;
+import oracle.weblogic.kubernetes.logging.LoggingFacade;
+import oracle.weblogic.kubernetes.utils.ExecResult;
+import org.awaitility.core.ConditionFactory;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static oracle.weblogic.kubernetes.TestConstants.ELASTICSEARCH_HOST;
+import static oracle.weblogic.kubernetes.TestConstants.ELASTICSEARCH_HTTP_PORT;
+import static oracle.weblogic.kubernetes.TestConstants.KIBANA_INDEX_KEY;
+import static oracle.weblogic.kubernetes.actions.TestActions.execCommand;
+import static oracle.weblogic.kubernetes.actions.TestActions.getOperatorPadName;
+import static oracle.weblogic.kubernetes.assertions.impl.Kubernetes.isPodReady;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.checkServiceExists;
+import static oracle.weblogic.kubernetes.utils.ThreadSafeLogger.getLogger;
+import static org.awaitility.Awaitility.with;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Utility class for ELK Stack and WebLogic logging exporter.
+ */
+public class LoggingExporter {
+
+  private static LoggingFacade logger = getLogger();
+  private static ConditionFactory withStandardRetryPolicy =
+      with().pollDelay(2, SECONDS)
+          .and().with().pollInterval(10, SECONDS)
+          .atMost(5, MINUTES).await();
+
+  private static final int maxIterationsPod = 10;
+
+  /**
+   * Install Elasticsearch.
+   *
+   * @param params the parameters to install Elasticsearch
+   * @return true if the Elasticsearch is successfully installed, false otherwise.
+   */
+  public static boolean installElasticsearch(LoggingExporterParams params) {
+    String elasticsearchName = params.getElasticsearchName();
+    String namespace = params.getLoggingExporterNamespace();
+    Map labels = new HashMap<String, String>();
+    labels.put("app", elasticsearchName);
+
+    // create Elasticsearch deployment CR
+    logger.info("Create Elasticsearch deployment CR for {0} in namespace {1}",
+        elasticsearchName, namespace);
+    V1Deployment elasticsearchDeployment = createElasticsearchDeploymentCr(params);
+
+    // create Elasticsearch deployment
+    logger.info("Create Elasticsearch deployment {0} in namespace {1}", elasticsearchName, namespace);
+    boolean deploymentCreated = assertDoesNotThrow(() -> Kubernetes.createDeployment(elasticsearchDeployment),
+        String.format("Create Elasticsearch deployment failed with ApiException for %s in namespace %s",
+          elasticsearchName, namespace));
+    assertTrue(deploymentCreated,
+        String.format("Create Elasticsearch deployment failed with ApiException for %s in namespace %s",
```

Comment:
change the message as if it failed with ApiException it is captured in the above line

Reply:
changed